### DISCUSSION TITLE
insert into table without identity primary key

### DISCUSSION THREAD

**Discussion by nickali2** [go-rel/rel#281] [Q&A] [answered]
how can i insert into a mssql table without identity primary key? 
how set identity on primary key OFF?

**Comment by Fs02** [comment, marked as answer]
do you mean the table it self doesn't have any primary key? or do you want to let database to populate it?

**Reply by nickali2** [reply]
let database to populate the key

**Reply by Fs02** [reply]
in that case, you only need to `Insert` as normally.
if you don't specify any value on your ID field, REL will let database to populate it

```golang
book := Book{
     // ID field is left blank (Zero value) because this will be populated by DB
    Title:    "Rel for dummies",
    Category: "education",
    Author: Author{
        Name: "CZ2I28 Delta",
    },
}

// Insert directly using struct.
err := repo.Insert(ctx, &book)
```
https://go-rel.github.io/crud/#create
